tools/downloader/topologies.py

```diff
@@ -0,0 +1,191 @@
+import sys
+import os
+import re
+import yaml
+import types
+import argparse
+
+from pathlib import Path
+
+import common
+
+_downloader = Path(__file__).parent / 'downloader.py'
+_precision_marker = '$precision'
+_cache = Path(os.getenv('OPENCV_OPEN_MODEL_ZOO_CACHE_DIR', 'models')).resolve()
+
+
+for _topology in common.load_topologies(argparse.Namespace(config=None)):
+    _name = _topology.name.replace('-', '_').replace('.', '_')
+    _files = _topology.files
+    _download_dir = _cache / _topology.subdirectory
+    _mo_args = {}
+    if _topology.mo_args:
+        for _arg in _topology.mo_args:
+            _tokens = _arg.split('=')
+            _mo_args[_tokens[0]] = _tokens[1] if len(_tokens) == 2 else None
+
+    # Resolve weights and text files names.
+    assert(len(_files) > 0)
+
+    if len(_files) > 2:
+        assert(_topology.framework == 'dldt'), ('Unexpected framework type: ' + _framework)
+
+        _paths = []
+        for i in range(2):
+            # Get only basename and add precision marker
+            _path = str(_files[i].name)
+            _pos = _path.find('/')
+            if _path[:_pos] in common.KNOWN_PRECISIONS:
+                _path = _precision_marker + _path[_pos:]
+            _paths.append(_path)
+
+        _config_path, _model_path = _paths
```

Comment:
This relies on the assumption that the first two files listed are an `.xml` file and a `.bin` file (in that order), which is too unreliable. What you should depend on is the fact that the models will always be saved to `<precision>/<name>.xml`/`.bin`. Which, admittedly, is still an assumption, but I think I'm going to add logic to `common.py` to enforce this soon, at which point it'll be a contract.

Reply:
@IRDonch, It's useful to pass parameters in correct order. We have a method in OpenCV with is independent to network files order. It can can determine origin framework and config/model paths depends on extensions.

I can add some similar logic here.

Reply:
Added reversed paths in case of swapped `.xml` and `.bin` files. Thanks!